enhancements/installer/gcp-dns-creation-specified-project.md

```diff
@@ -0,0 +1,223 @@
+- Provisioning and/or configuring the public DNS zone. When the user specifies the
+  project for the private DNS Zone, the public zone must exist in the service project
+  with the correct base domain.
+- Provisioning and/or configuring the Service Accounts for the Installer and
+  Cluster are to be done manually by the user prior to running the installer.
+
+## Proposal
+
+The installer should be able to install a GCP IPI cluster using Shared VPC
+networking with a specified project and zone for private dns zone management
+by automating the majority of the current GCP UPI workflow. This
+requires some resources to be created and configured in advance, such as the
+DNS public and private zones. The installer, and the cluster-ingress-operator, will
+be updated to enable some of this functionality. They will become capable of
+creating and managing resources that exist in a different (Service) project from
+the cluster (Service and/or Host Project).
+
+### Workflow Description
+
+Prior to executing the installer, the Shared VPC, Network, and Subnets will be
+manually created in the Host Project and configured for Shared VPC (XPN)
+configuration with the Service Project. A service account will also be created
+in the Service Project and granted sufficient permissions in the Host Project.
+The DNS zones _may_ be created in a separate Service Project. When a private DNS
+zone is specified in the installconfig file, the public zone must exist in that same
+project with the base domain found in the installconfig file. The private DNS zone 
+provided by the user does **not** have to exist prior to creation, the specified name
+will be used during installation, and if the zone does not exist one will be created 
+using the matching name. A service account will also be created in the Second Service 
+Project and granted sufficient permissions in the Host project.
+
+After all of these resources are created, they will be used in a manually
+created install-config.yaml to configure the cluster accordingly. At a minimum,
+the credentialsMode will be set to Passthrough, and the network,
+computeNetwork, controlPlaneNetwork, and networkProjectID will be configured to
+reference the Shared VPC network. When minimial permissions are desired in the
+Host Project, the additional parameters will be configured to enable the
+cluster to consume them as well.
+
+The installation process can then proceed as any other IPI workflow by running
+the create manifests, create ignition, and/or create cluster sub-commands.
+
+### API Extensions
+
+#### installconfig.platform.gcp.privateDNSZone
+
+A new parameter will be added to the install-config to specify the private DNS
+managed zone. This parameter will be used where necessary to validate existing
+DNS resources as well as provision additional private DNS records. The default
+will be an empty value, which will indicate these resources are to be created
+and managed in the cluster project identified by the projectID.
```

Comment:
I think the installconfig API is a bit different now, right? It should be `platform.gcp.dns.privateZone`?

```
$ ./bin/openshift-install explain installconfig.platform.gcp.dns.privateZone
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <object>
  PrivateZone contains the information for a private DNS zone. The Private DNS Zone can
only be supplied during Shared VPC (XPN) installs. The PrivateZone can exist or be
created in a second service project; a project other than the one matching projectID
or networkProjectID.

FIELDS:
    name <string> -required-
      Name is the name of the dns-managed zone.

    projectID <string>
      ProjectID is the project where the zone resides.
```